src/main/java/classify/data/DataReader.java

```diff
@@ -205,14 +205,60 @@ public static void initialiseData(ArrayList<Student> list,
             DataUI.printRetrieveData();
 
             //@@author ParthGandhiNUS
-            while (line.ready()) {
-                restoreStudentList(line.readLine(), list);
+            try {
+                while (line.ready()) {
+                    restoreStudentList(line.readLine(), list);
+                    line.close();
+                }
+            //@@author alalal47
+            } catch (ArrayIndexOutOfBoundsException e) {
+                DataUI.printFileCorruptedMessage();
+                DataUI.printDataDeletionPromptMessage();
+                promptDataDeletion();
+                UI.printDivider();
+            } catch (IOException e) {
+                DataUI.printFileCorruptedMessage();
+                DataUI.printDataDeletionPromptMessage();
+                promptDataDeletion();
+                UI.printDivider();
```

Comment:
Good use of the UI class to print out messages that corresponds with the corrupted information